porcupine/plugins/filetypes.py

```diff
@@ -228,10 +229,27 @@ def on_path_changed(tab: tabs.FileTab, junk: object = None) -> None:
     apply_filetype_to_tab(get_filetype_for_tab(tab), tab)
 
 
+def after_save(tab: tabs.FileTab, junk: object) -> None:
+    if tab.path == Path(dirs.user_config_dir) / "filetypes.toml":
+        # Reload all filetypes when the user saves filetypes.toml.
+        # If user has chosen a custom filetype, leave the tab alone.
```

Comment:
Why not just find a filetype with whatever `filetype_name` is currently used? It still needs a fallback to `get_filetype_for_tab()` in case the user deletes a filetype.

Reply:
If the user hasn't selected a custom filetype, we should call `get_filetype_for_tab()`, because it is possible that `filename_patterns` was modified.